### PR TITLE
Update release's state via admin area

### PR DESCRIPTION
**Pull request by hedgehog1833**
Should be merged after https://github.com/MetalDetectorRocks/metal-release-butler/pull/55

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Improve Data Quality](https://trello.com/c/u6D09MbB/222-improve-data-quality)

**Comment by DanielW1987** [comment]
I'll wait with the review until you have implemented the butler adjustments.

**Comment by hedgehog1833** [comment]
@DanielW1987 Everything ready now :)

File: butler/src/main/java/rocks/metaldetector/butler/client/ReleaseButlerRestClient.java [hunk 1]
```diff
@@ -18,4 +18,6 @@
 
   List<ButlerImportJob> queryImportJobResults();
 
+  void updateReleaseState(long  releaseId, String state);
```
Comment (DanielW1987) [inline butler/src/main/java/rocks/metaldetector/butler/client/ReleaseButlerRestClient.java:21]:
```suggestion
  void updateReleaseState(long releaseId, String state);
```

File: butler/src/main/java/rocks/metaldetector/butler/client/ReleaseButlerRestClientMock.java [hunk 2]
```diff
@@ -68,4 +68,8 @@ public void createRetryCoverDownloadJob() {
         new ButlerImportJob(684, 105, LocalDateTime.of(2020, 7, 7, 21, 14, 6), LocalDateTime.of(2020, 7, 7, 21, 19, 46), "Metal Achives")
     );
   }
+
+  @Override
+  public void updateReleaseState(long releaseId, String state) {
+  }
```
Comment (DanielW1987) [inline butler/src/main/java/rocks/metaldetector/butler/client/ReleaseButlerRestClientMock.java:74]: maybe you can add `log.info()` like the other methods

File: webapp/src/main/resources/static/js/admin/releases.js [hunk 3]
```diff
@@ -109,3 +116,70 @@ function resetUpdateReleaseForm() {
   $("#update-release-form")[0].reset();
   resetValidationArea('#release-validation-area');
 }
+
+/**
+ * Updates a release's state
+ */
+function updateRelease() {
+  const pathParam = $("#release-id").val();
+  $.post({
```
Comment (DanielW1987) [inline webapp/src/main/resources/static/js/admin/releases.js:125]: Maybe a `$.ajax` is less misleading (post vs. PUT).

File: webapp/src/main/resources/static/js/admin/releases.js [hunk 4]
```diff
@@ -109,3 +116,70 @@ function resetUpdateReleaseForm() {
...
+
+/**
+ * Success callback for updating a release's state.
+ */
+function onUpdateReleaseSuccess() {
+  releaseTable.rows().every(function (rowIndex) {
+    if (releaseTable.cell(rowIndex, 7).data() === parseInt($("#release-id").val())) {
+      releaseTable.cell(rowIndex, 6).data($('#release-state').val());
+    }
+  }).draw();
```
Comment (DanielW1987) [inline webapp/src/main/resources/static/js/admin/releases.js:157]: Is there no better solution than iterating over 1000 items first? You only have to take the Row-Index with you when you open the Udate-Form? You can pass this index right up to here.

Maybe something like `let data = releaseTable.row(this).index();` in function `showUpdateReleaseForm()`.

See <https://datatables.net/reference/api/row().index()>.

Comment (hedgehog1833) [reply webapp/src/main/resources/static/js/admin/releases.js:157]: As discussed I use a hidden input field to pass the currentRowIndex.

**Review by DanielW1987 — APPROVED** [review]

**Pull request merged**